lib/src/message_parser.dart

```diff
@@ -180,17 +189,40 @@ class _ParserUtil {
 
   At<LiteralString>? nonIcuMessageText(int at) {
     if (at < input.length) {
-      var match = nonOpenBracketRegex.matchAsPrefix(input, at);
-      if (match != null) {
-        var matchGroup = match.group(0);
-        if (matchGroup != null) {
-          return At(LiteralString(matchGroup), match.end);
-        }
+      var inputAt = input.substring(at);
+      var indexOf = indexOfUnescapedOpenBracket(inputAt);
+      String matched;
+      if (indexOf == 0) {
+        return null;
+      } else if (indexOf == -1) {
+        matched = inputAt;
+      } else {
+        matched = inputAt.substring(0, indexOf);
       }
+      return At(LiteralString(matched), at + matched.length);
     }
     return null;
   }
 
+  int indexOfUnescapedOpenBracket(String inputAt) {
+    const quote = 39; // Character '
```

Comment:
One pattern I've seen with char constants in dart is to use a `$` prefix to the name; perhaps to help call it out as a special constant? So,

```
const int $space = 0x20;
```

Reply:
I'd move these char constants to somewhere were you can share them between here and `lib/src/messages/message.dart`.